src/groupByKey.ts

```diff
@@ -2,6 +2,22 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends object>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  const groupedMap: GroupsMap<T> = {};
+
+  items.map((item) => {
```

Comment:
See how you ignored `.map`'s return and used the external object `groupedMap` to store each step's result? This is a clear indicator that you're using the wrong array function!

`.reduce` would be more fitting to this problem